third_party/terraform/resources/resource_google_folder_organization_policy.go

```diff
@@ -51,12 +51,16 @@ func resourceFolderOrgPolicyImporter(d *schema.ResourceData, meta interface{}) (
 }
 
 func resourceGoogleFolderOrganizationPolicyCreate(d *schema.ResourceData, meta interface{}) error {
+	d.SetId(fmt.Sprintf("%s:%s", d.Get("folder"), d.Get("constraint")))
```

Comment:
Setting the ID before the error is going to be a change in behavior. Before if there was a bad config that resulted in an error the next `apply` would look like it was trying to create the resource again. However constraint+policy will never 404, it will instead return a default representation of the policy. This would now mean that if there was an error a subsequent `apply` would look like the resource had been partially created and an update would be issued.

IMO this is an ok change as either way the user will get an error and need to re-run apply to get to the desired state. The only difference is that if the resource was removed it would require a `destroy` call, but the result from a failed create and a destroy would still be the same.